boards/microbit_v2/src/main.rs

```diff
@@ -522,6 +527,19 @@ pub unsafe fn main() {
         512
     ));
 
+    //Nonvolatile_storage
+    let nonvolatile_storage = components::nonvolatile_storage::NonvolatileStorageComponent::new(
+        board_kernel,
+        capsules::nonvolatile_storage_driver::DRIVER_NUM,
+        &base_peripherals.nvmc,
+        0x00044000, // Start address for userspace accessible region
+        0x0003C000, // Length of userspace accessible region
+        0x00000000, //start address of kernel region
+        0x00040000, // length of kernel region
+    )
+    .finalize(components::nv_storage_component_helper!(
+        nrf52833::nvmc::Nvmc
+    ));
```

Comment:
It's probably worth splitting this PR into a few separate ones. That way it can be clear what each one is doing and why

Reply:
As I mentioned in the pull request overview, I left nonvolatile_storage section. If necessary, I will separate it from my pull request.